### PR TITLE
Add an error testing feature in controller

### PR DESCRIPTION
**Pull request by rochefort**
refer to https://github.com/fmalk/codeigniter-phpunit

I want to test 404 and 500 requests.

example:
```php
	/**
	* @expectedException		   PHPUnit_Framework_Exception
	* @expectedExceptionCode	   404
	*/
	public function test_index()
	{
		$output = $this->request('GET', ['nocontroller', 'noaction']);
	}
```

What do you think this aproach?
Please let me know if you know of any better way.

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you. Merged: https://github.com/kenjis/ci-phpunit-test/commit/70a487439c1eaf9f291a0308ecede57f28f7e52b

I only fixed white spaces.

**Comment by kenjis** [comment]
Updated README: https://github.com/kenjis/ci-phpunit-test/commit/262b1c8b622cd0d501a9dc9bf4a8877299830eb6

**Comment by kenjis** [comment]
I forgot to answer your question.

If you want to test 404 requests, I think this approach is the best for now.
We can't handle `exit()`, so we must remove it in testing.

I hope CodeIgniter core handle them and I proposed to remove all exit() and add throwing Exceptions.
But it was rejected.

**Comment by rochefort** [comment]
Thank you for your merging

> But it was rejected.

Oh, I’ve got it. It's a shame.

File: application/tests/TestCase.php [hunk 1]
```diff
@@ -235,4 +240,41 @@ public function warningOn()
...
+
+		require_once(APPPATH.'controllers/'.$RTR->directory.$class.'.php');
+		if ( ! class_exists($class, FALSE) OR $method[0] === '_' OR method_exists('CI_Controller', $method))
+		{
+			return TRUE;
+		}
+
+		if (method_exists($class, '_remap'))
+		{
+			$params = array($method, array_slice($URI->rsegments, 2));
```
Comment (kenjis) [inline application/tests/TestCase.php]: I'm sorry, I overlooked this.

`$URI` is not defined.

Comment (rochefort) [inline application/tests/TestCase.php]: oops! :bomb: I'm so sorry.
I think that is not necessary, so I remove that.
How about you?

https://github.com/kenjis/ci-phpunit-test/commit/cca05807c6ace30aa5933bfd5e7ccc7239237888

**Comment by kenjis** [comment]
I'm sorry.
I've been completely confused request to controller method and URI string (routing).

URI means a part of URL. 
For example:
~~~
URL: http://localhost/codeigniter/index.php/abc/def/xyz
URI: /abc/def/xyz
~~~

My intention is request to controller method, not URI string:
https://github.com/kenjis/ci-phpunit-test/blob/master/docs/FunctionAndClassReference.md#testcaserequestmethod-argv-params---callable--null

So I should not care about routing at all. But my code was wrong, because I used `CI_Router` object.

I rewrote `request()` method, and separated it into two methods internally.
1. request to controller (`callControllerMethod`)
2. request to URI (`requestUri`)

If there is something wrong, feel free to inform me.

**Comment by rochefort** [comment]
I apologize for making you confused.
Thank you for supporting `URI string`. I think it's a best implementation.

**Comment by kenjis** [comment]
No, you don't have to apologize. It's my fault.
Thank you for your feedback!

**Pull request closed without merging**